src/FakeItEasy/OutAndRefParametersConfigurationExtensions.cs

```diff
@@ -34,15 +35,16 @@ public static class OutAndRefParametersConfigurationExtensions
         /// Any non out and ref parameters are ignored.
         /// The function will be called each time this call is made and can return different values.
         /// </summary>
+        /// <typeparam name="TInterface">The type of configuration interface to return.</typeparam>
         /// <typeparam name="T1">Type of the first argument of the faked method call.</typeparam>
         /// <param name="configuration">The configuration to extend.</param>
         /// <param name="valueProducer">A function that produces the output values.</param>
         /// <returns>A configuration object.</returns>
         /// <exception cref="FakeConfigurationException">
         /// The signatures of the faked method and the <paramref name="valueProducer"/> do not match.
         /// </exception>
-        public static IAfterCallSpecifiedConfiguration AssignsOutAndRefParametersLazily<T1>(
-            this IOutAndRefParametersConfiguration configuration, Func<T1, object[]> valueProducer)
+        internal static IAfterCallSpecifiedConfiguration<TInterface> AssignsOutAndRefParametersLazily<TInterface, T1>(
+            this IOutAndRefParametersConfiguration<TInterface> configuration, Func<T1, object[]> valueProducer)

```

Comment:
The addition of the `TInterface` type param made this method, and the following ones, too impractical to use, so they had to be moved to the `IOutAndRefParametersConfiguration<TInterface>` interface. The extension methods are still used by the classes that implement the interface, to avoid code duplication.